### PR TITLE
fix non-editable views state issue and conditional row actions for TableView

### PR DESCRIPTION
**Pull request by imanjra**
## What changes are proposed in this pull request?

fix non-editable views state issue and conditional row actions for TableView

## How is this patch tested? If it is not, please explain why.

Using a python panel

### Is this a user-facing change that should be mentioned in the release notes?

-   [x] No. You can skip the rest of this section.
-   [ ] Yes. Give a description of this change to be included in the release
        notes for FiftyOne users.

(Details in 1-2 sentences. You can just refer to another PR with a description
if this PR is part of a larger change.)

### What areas of FiftyOne does this PR affect?

-   [ ] App: FiftyOne application changes
-   [ ] Build: Build and test infrastructure changes
-   [ ] Core: Core `fiftyone` Python library changes
-   [ ] Documentation: FiftyOne documentation changes
-   [ ] Other

## Summary by CodeRabbit

- **New Features**
	- Introduced a new utility function to enhance schema handling.
	- Added a constant to define non-editable view names.
	- New property added to customize the selected cell's background color in the table component.
- **Improvements**
	- Refactored row actions in the TableView component for dynamic retrieval based on the current row.
	- Enhanced logic for state initialization in the DynamicIO component, ensuring it only initializes for editable schemas.
- **Style**
	- Minor adjustments to the formatting and alignment of table cells.

### CODE REVIEW

**Comment by imanjra** [comment]
Converting to draft to fix highlight colouring

File: app/packages/core/src/plugins/SchemaIO/utils/index.ts [hunk 1]
```diff
@@ -92,6 +92,27 @@ const COMPOSITE_VIEWS = [
   "ButtonGroupView",
 ];
 
+const NON_EDITABLE_VIEWS = [
```
Comment (ritch) [inline app/packages/core/src/plugins/SchemaIO/utils/index.ts:95]: This seems hard to maintain. I would rather we define this in each view somehow or in a way that we won't forget to add things to this list. In other words I don't think we'll be able to maintain this list.

Comment (imanjra) [reply app/packages/core/src/plugins/SchemaIO/utils/index.ts:95]: Yah, good point. I will follow-up. I would say let's go with this for now to unblock Minh's panel for release

**Review by minhtuev — APPROVED** [review]
Bug fix works for TableView, verified here: 
https://www.loom.com/share/581654042f7e42b192beafe9ace62d09

Thanks Ibrahim! ✅

**Pull request merged**